docs/doxygen/Doxymain.md

```diff
@@ -0,0 +1,76 @@
+# Welcome to p4 Compiler's documentation!
+
+@htmlonly
+<div class="diagram-container">
+<!-- draw.io diagram -->
+<div class="mxgraph" style="" data-mxgraph="{&quot;highlight&quot;:&quot;#0000ff&quot;,&quot;lightbox&quot;:false,&quot;nav&quot;:true,&quot;edit&quot;:&quot;_blank&quot;,&quot;url&quot;:&quot;https://drive.google.com/uc?id=16X8Z_4-J8mHQPxuxm9HjWppPMTTXZdZ6&amp;export=download&quot;}"></div>
+<script type="text/javascript" src="https://viewer.diagrams.net/embed2.js?&fetch=https%3A%2F%2Fdrive.google.
+com%2Fuc%3Fid%3D16X8Z_4-J8mHQPxuxm9HjWppPMTTXZdZ6%26export%3Ddownload"></script>
+</div>
+
+<button id="toggle" onclick="animateFlow()">Toggle</button>
+
+<script>
+    function animateFlow() {
+        const svgDocument = document.querySelector('.mxgraph svg');
+        if (svgDocument) {
+            const edges = svgDocument.querySelectorAll('path:not([marker-end])');
+            edges.forEach(edge => {
+                edge.classList.add('flow');
+            });
+            const toggleButton = document.getElementById('toggle');
+            if (toggleButton) {
+                toggleButton.style.display = 'none';
+            }
+        }
+    }
+</script>
+@endhtmlonly
+
+Join the future of programmable networking with **p4c**, a tool that is transforming the way we handle network data. As networks evolve, so does the need for flexible and efficient solutions to manage increasing volumes of data. **p4c** stands at the forefront of this evolution, offering a revolutionary approach to network programming.
+
+`p4c` is a open source reference compiler for the [P4 (Programming Protocol-independent Packet Processors)](https://p4.org/) programming language. It supports both P4-14 and P4-16 versions, making it a versatile tool for developers looking to leverage the power of programmable networking.Remarkably, the front-end, mid-end, and P4C-graphs backend of p4c serve as the foundation for at least one commercially supported P4 compiler, testament to its robustness and reliability.
```

Comment:
```suggestion
P4C is the official open-source reference compiler for the [P4 (Programming Protocol-independent Packet Processors)](https://p4.org/) programming language. It supports both P4-14 and P4-16.
```

I would look at Section 3 this paper for example https://www.usenix.org/system/files/osdi20-ruffy.pdf on an alternative description of the P4 compiler. We can pick the parts that are relevant.